homeassistant/components/energyzero/services.py

```diff
@@ -75,12 +82,29 @@ def __serialize_prices(prices: Electricity | Gas) -> ServiceResponse:
     }
 
 
+async def __get_coordinator(
+    hass: HomeAssistant, call: ServiceCall
+) -> EnergyZeroDataUpdateCoordinator:
+    entry_id: str = call.data[ATTR_CONFIG_ENTRY]
+    entry: ConfigEntry | None = hass.config_entries.async_get_entry(entry_id)
+
+    if not entry:
+        raise HomeAssistantError(f"Invalid config entry: {entry_id}")
```

Comment:
Nowadays we want to raise `ServiceValidationError` for bad service input.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/

Add translation strings for the exception message.